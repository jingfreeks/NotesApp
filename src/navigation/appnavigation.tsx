import React from 'react';
import bottomtab from './bottomtabnavigation';
import {Navigation} from './constant';
import {Notes, NotesMap} from '@/screens';
import type {AppNavigationProps, RootNavigationProps} from './types';
import {createStackNavigator} from '@react-navigation/stack';

const Stack = createStackNavigator<RootNavigationProps & AppNavigationProps>();

export const Appnavigation = () => {
  return (
    <Stack.Navigator>
      <Stack.Screen
        component={bottomtab}
        name={Navigation.Bottomtab}
        options={{headerShown: false}}
      />
      <Stack.Screen
        component={Notes}
        name={Navigation.Notes}
        options={{
          headerBackTitleVisible: false,
          headerShown: true,
          title: 'Notes Form',
          headerTitleStyle: {
            fontWeight: 'bold',
          },
        }}
      />
      <Stack.Screen
        component={NotesMap}
        name={Navigation.NotesMap}
        options={{
          headerBackTitleVisible: false,
          headerShown: true,
          title: 'Notes Map',
          headerTitleStyle: {
            fontWeight: 'bold',
          },
        }}
      />
    </Stack.Navigator>
  );
};
