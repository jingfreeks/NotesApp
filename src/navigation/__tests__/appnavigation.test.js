import React from 'react';
import { renderWithProviders } from "@/utils/testframework";
import {Appnavigation} from '../appnavigation';


describe('App navigation Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Appnavigation />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
