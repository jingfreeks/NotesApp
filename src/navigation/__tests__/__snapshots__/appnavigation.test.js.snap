// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App navigation Screen Should work as expected to get snapshot 1`] = `null`;
